docs/structured-streaming-programming-guide.md

```diff
@@ -1814,6 +1814,82 @@ Specifically for built-in HDFS state store provider, users can check the state s
 it is best if cache missing count is minimized that means Spark won't waste too much time on loading checkpointed state.
 User can increase Spark locality waiting configurations to avoid loading state store providers in different executors across batches.
 
+### State Store
+
+State store is a versioned key-value store which provides both read and write operations. In
+structured streaming, we use the state store provider to handle the state store operations crossing
+batches. There are two build-in state store provider implementations. End users can also implement
+their own state store provider by extending StateStoreProvider interface.
+
+#### HDFS state store provider
+
+The HDFS backend state store provider is the default implementation of [[StateStoreProvider]] and
+[[StateStore]] in which all the data is backed by files in an HDFS-compatible file system. All
```

Comment:
Should we also mention the data is stored in memory map in executors?

Reply:
Make sense, we mentioned this drawback in the RocksDB one, but it should also be mentioned here.